deployment-repository/src/main/java/org/jboss/as/repository/ContentRepository.java

```diff
@@ -365,18 +366,23 @@ public void removeContent(ContentReference reference) {
                 Path file = getDeploymentContentFile(reference.getHash(), true);
                 try {
                     Files.deleteIfExists(file);
+                    assert !Files.exists(file);
```

Comment:
Just a note for posterity.  Well, a couple notes.

First, exists and notExists are not strictly opposites.  One determines if the file definitely exists, and one determines if the file definitely does not exist.  If the existence of a file cannot be determined, notExists and exists will both return false.

Second, this is inherently racy as the file could theoretically be recreated between the delete and the check.  Yes, it's unlikely, but unlikely things have a way of becoming likely in production environments.